be/src/formats/orc/orc_chunk_reader_factory.h

```diff
@@ -0,0 +1,197 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <orc/OrcFile.hh>
+
+#include "formats/orc/orc_row_reader_filter.h"
+#include "formats/orc/orc_chunk_reader.h"
+#include "common/status.h"
+
+namespace starrocks {
+
+class OrcChunkReaderFactory {
+public:
+    virtual StatusOr<std::unique_ptr<OrcChunkReader>> create() = 0;
+
+    void set_src_slot_descriptors(std::vector<SlotDescriptor*> src_slot_descriptors) {
```

Comment:
pass `set_xxx` as a argument of ctor, so caller of won't forget to set them.

Reply:
fixed